File: app/src/main/java/com/sanketvagadiya/Constants/GetOnlineData.java
package com.sanketvagadiya.Constants;

import android.os.AsyncTask;
import android.util.Log;

import org.json.JSONException;
import org.json.JSONObject;


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;

public class GetOnlineData extends AsyncTask<Void, Void, JSONObject> {

    String url = "";
    String jsonStr;

    JSONObject jsonObject;

    public GetOnlineData(String url) {
        this.url = url;
    }


    @Override
    protected JSONObject doInBackground(Void... voids) {

        try {
            URL url = new URL(this.url);
            HttpURLConnection httpURLConnection = (HttpURLConnection) url.openConnection();
            InputStream inputStream = httpURLConnection.getInputStream();
            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));

            jsonStr = reader.readLine();

            jsonObject = new JSONObject(jsonStr);
            Log.e("jsonObj GetOnlineData", "." + jsonObject.toString());

        } catch (MalformedURLException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return jsonObject;
    }

}

File: app/src/main/java/com/sanketvagadiya/Constants/GetOnlineData.kt
package com.sanketvagadiya.Constants

import android.os.AsyncTask
import android.util.Log

import org.json.JSONException
import org.json.JSONObject


import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL

class GetOnlineData(url: String) : AsyncTask<Void, Void, JSONObject>() {

     var url = ""
     var jsonStr= ""

     lateinit var jsonObject: JSONObject

    init {
        this.url = url
    }


    override fun doInBackground(vararg voids: Void): JSONObject {

        try {
            val url = URL(this.url)
            val httpURLConnection = url.openConnection() as HttpURLConnection
            val inputStream = httpURLConnection.inputStream
            val reader = BufferedReader(InputStreamReader(inputStream))

            jsonStr = reader.readLine()

            jsonObject = JSONObject(jsonStr)
            Log.e("jsonObj GetOnlineData", ".$jsonObject")

        } catch (e: MalformedURLException) {
            e.printStackTrace()
        } catch (e: IOException) {
            e.printStackTrace()
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return jsonObject
    }

}
